src/posts/2020-07-30-the-worst-decision-of-my-career.md

```diff
@@ -0,0 +1,131 @@
+
+The idea is that a complex system like our body uses a sort of "message passing" to function. Plus, if one part of the system fails, the others keep going: if you lose sensitivity in your hands, you can still move them, because a different system handles that. If our body, which is a fantastic piece of biology, works like this, then this must be the best way to make a resilient, adaptative, piece of software!
+
+This analogy can impress a crowd, but there are many flaws to it. The first is that the human body evolved from unicellular organisms. It wasn't designed as it is, so it had to go through many phases to reach what we got now.
+
+The second issue with this is that the adaptations the human body got aren't even the best ones; they are the ones that were good enough and showed at the right time.
+
+I can stay here all day, trashing this analogy because it isn't a very good one. My point is, this analogy can be used to explain how an event-driven system and microservices work, but not to prove the human body is an example of why even-driven is the way to go (yes, I've heard this argument).
+
+There are two things that you must take into account if you're going with microservices and event-driven architecture. The first is that you need to have a strong DevOps culture. Microservices will make a lot of things more complicated, especially your infrastructure. So be mindful if you want to be spending your time in the infrastructure or the product. The second is team size. Your system will have a lot more layers, making it harder to change and reason about. Bigger teams with well enough defined scopes will benefit more from this.
+
+### Accounting
+
+For event-sourcing, the analogy is usually around accounting. Accountants don't use erasers, amiright? For thousands of years, we invented and evolved mechanisms to keep track of things. Double-entry accounting was invented to guarantee accurate financial records because it reveals data entry errors. _This is awesome, so let's do the same thing in code now! Besides, if you never delete anything, you can always go back and look, and maybe use that data for other purposes._
+
+Slow down! Accountants invented systems around manual work. Those systems were designed to prevent human error. We still have to account for human error in software, but it's a different kind of error, so we need a different kind of solution.
+
+Event-sourcing indeed solves a lot of problems, but there are other, simpler, solutions that you can use. You can keep track of the changes in a table (on your database) with just another table and some triggers. If you're into Rails, [paper_trail](https://github.com/paper-trail-gem/paper_trail) is the tool for you.
+
+Is your data time-series? No problem. Put those events to a table and use a materialized view to read the data. Small steps can take you a long way.
+
+## Banking
```

Comment:
Are you sure this and `Making the right call` have the correct heading level?

Reply:
True! Somehow I messed this up again.